### ISSUE TITLE
Make it possible to change ThriftTask's source directory

### ISSUE THREAD

**Issue by SolalPirelli** [thomaslee/gradle-thrift-plugin#2]
First, thanks for this plugin, it's really useful. :)

For various reasons, I need to compile Thrift sources to multiple languages independently, e.g. sometimes I must only compile to Java and sometimes only to Objective-C. So I want to create multiple tasks of type `ThriftTask`, and not use the default `generateThriftSource`.

I tried creating a task of type `ThriftTask`, but it's always skipped because of a lack of source files. I think it's because `ThriftTask` defines a `setSource(SourceDirectorySet)` and only uses that as its source, not the `source` protected field defined by its parent `SourceTask` (and set by `setSource(Object)`, which means e.g. `source 123` runs fine but is actually useless within a `ThriftTask`). `SourceDirectorySet` is only implemented by `DefaultSourceDirectorySet`, an internal Gradle class that's not easy to use from a build script since it requires a `FileResolver`, another internal Gradle interface that must be injected by Gradle.

TL;DR: **Could you please allow build scripts to change the source directory?** This is (I think) the only required feature to be able to create tasks of type `ThriftTask` that work properly. Thanks!
